### ISSUE TITLE
Add support for Tobii Communicator

### ISSUE THREAD

**Issue by gavinhenderson** [Baton-donation/app#4]
We want to be able to extract history from [Tobii Communicator](https://uk.tobiidynavox.com/products/communicator-5) so that we can use the data to support our data collection.

We have found that there exists a `.phr` file called `Speech history.phr` that we can use to get the speech history from but we don't know how to full parse it.

You can see an example history file here: https://github.com/AceCentre/AAC-Corpora-Collecting/blob/main/Communicator/Phrases/Speech%20history.phr

If I parse the file with UTF-16 then it looks like:

```
﻿⯿Hello how are you I think it's over cooked.䉀﻿ᇿHello how are you䉀﻿׿Hello䉀﻿⃿Hello, my name is this low carb?䉀﻿᛿Hi, this is delicious!䉀
```

I suspect that the characters it doesn't know are control characters of some kind. It would be nice to know what the control characters are so we can parse it nicely.

I'm going to send this issue to folks at Tobii and see if they can shed any light on how we could parse the `.phr` file or if there is a different and better way to get speech history

**Comment by codetheweb** [comment]
I'm guessing they're end-of-sentence / input markers. In hex:

```
0x40420F00FFFEFF
```

Dasher has something similar. But would be good to get confirmation from Tobii. 👍

**Comment by joedevsys** [comment]
This should get you started, at least as an initial guess:

```
struct PhraseFile{
    uint32  unknown;    //could be version or file type identifier seems to always be 0x02000000
    uint32  history_flag;  //seems to be 1 in speech history.phr, other files seems to be 0
    uint32  num_of_entries;
    Phrase phraseList[num_of_entries];
}

struct Phrase{
    uint24 separator;    //<0xfffeff> even with string length includede it's useful to have a flag so you can scan the list backwards.
    uint8 string_len;
    uint16 string[string_len];             
    uint32 unknown2;       //seems to be 0x40420f00 in speech history.phr and 0x00000000 in other files
}
```

Of course the uint32's could be more complex e.g. uint16; uint16;  etc

**Comment by joedevsys** [comment]
I have some .phr files that are not history files (i.e. static phrase stores) where a small number of unknown2 values are 0x01000000 or 0x03000000, but majority are still 0x00000000.  It's interesting to observe that  0x40420f00 in speech history.phr is little-endian for decimal 1000000. Treating it as uint32 produces this nice round number which doesn't happen if you try treating it as uint16 etc - of course this may be a red herring!

**Comment by gavinhenderson** [comment]
Thanks @joedevsys thats really helpful, thanks for doing so much digging about this stuff! 

Hopefully knowing all these control characters we should be able to extract the sentences.

**Comment by codetheweb** [comment]
Closed by v0.8.0.
